lib/collection/proxy-config.js

```diff
@@ -85,9 +106,22 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
         if (!_.isObject(options)) {
             return;
         }
-        _.has(options, 'server') && this.server.update(options.server);
-        _.isString(options.match) && (this.match = new UrlMatchPattern(options.match));
-        _.has(options, 'server') && (this.tunnel = _.isBoolean(options.tunnel) ? options.tunnel : false);
+        _.isString(options.host) && (this.host = options.host.replace(regexes.protocolMatcher, ''));
+        _.isInteger(options.port) && (options.port >= 0 && options.port <= 65535) && (this.port = options.port);
+        _.isBoolean(options.tunnel) && (this.tunnel = options.tunnel);
+
+        // match pattern
+        if (_.isString(options.match)) {
+            if (_.isEmpty(options.match) || options.match === MATCH_ALL_URLS) {
+                this.match = new UrlMatchPattern();
+            }
+            else {
+                this.match = new UrlMatchPattern('://' + options.match.replace(regexes.protocolMatcher, ''));
```

Comment:
Shall we have the '://' handling in the UrlMatchPattern itself? which behaves differently, if the protocols are provided seperately?